src/test/kotlin/calculator/ExpressionFactoryTest.kt

```diff
@@ -0,0 +1,42 @@
+package calculator
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.StringSpec
+
+class ExpressionFactoryTest : StringSpec({
+    val expressionFactory = ExpressionFactory(
+        object : SeparationStrategy {
+            override fun separate(input: String): List<String> {
+                return input.split(" ")
```

Comment:
```suggestion
                return input.split(OneSpaceSeparationStrategy.SEPARATOR)
```
OneSpaceSeprationStrategy에 정의한 SEPARATOR를 활용해보면 어떨까요?

Reply:
테스트에서 다른 클래스를 최대한 개입하지 않도록 구현하는 것에 목적을 두었던 것 같습니다 :)
`OneSpaceSeprationStrategy`에 변경이 생기면 해당 테스트가 영향을 받는 상황이 생길텐데 최대한 의존하지 않는 형태로 가는 것이 괜찮지 않을까요 ?

Reply:
가급적이면 실제 로직의 변경이 테스트 코드에 영향이 가도록 작성하는 것이 좋습니다.
이렇게 하면 프로덕트 코드 변경으로 인한 영향도를 쉽게 파악할 수 있습니다. (여기를 변경했더니 관련 테스트코드가 이것이고 이 테스트가 깨지는구나)

가령 빈 문자열을 두 칸으로 변경하게 된다면, 실제 코드와 테스트 코드 둘 다 변경해야할 것 같은데요.
만약 빈 문자열이 하나든 두개든 모두 하나로 취급하는 로직이 적용되어 있는 상태에서, 프로덕션 코드에는 빈 문자열이 두칸으로 되어 있고 테스트 코드에는 빈 문자열이 한칸으로 되어 있다면, 실제 프로덕트 코드와 테스트 코드에 적용되어 있는 seperator가 다른데 테스트 코드는 통과하지 않을까요? 🤔

Reply:
> 가급적이면 실제 로직의 변경이 테스트 코드에 영향이 가도록 작성하는 것이 좋습니다.
이렇게 하면 프로덕트 코드 변경으로 인한 영향도를 쉽게 파악할 수 있습니다. 

리뷰어님이 이야기해주신 이 부분이 많이 공감이 되는 것 같습니다 :)
다만 전략패턴으로 분리해놓은 부분의 구현체에서 코드의 변경이 생겼을 때, 전략 패턴을 사용하는 클래스의 테스트까지 영향을 미치도록 하는 것이 좋을지에 대해서는 조금 고민이 되는 것 같습니다.

ExpressionFactory에서 전략 패턴을 사용하여 인터페이스로 분리하여 의존성을 분리하였는데 구현체를 테스트에서 사용하게 될 경우 다시 ExpressionFactory가 전략 패턴의 구현체에 의존하는 상황이 된다고 생각하는데요.

문자열 분리 정책의 변경과 같은 코드 변경으로 인한 영향도는 전략 패턴의 구현체의 테스트(OneSpaceSeparationStrategyTest)로 파악하고, ExpressionFactory에서는 전략패턴에 의존하지 않는 부분을 담당하게 하는 것이 좀 더 작은 단위테스트를 구현할 수 있지않을까요?